File: backend/src/ZealEducation.Domain/Entities/FeeStructure.cs
using ZealEducation.Domain.Common;
using ZealEducation.Domain.Enums;

namespace ZealEducation.Domain.Entities;

public class FeeStructure : BaseAuditableEntity
{
    public Guid CandidateId { get; set; }
    public decimal TotalFee { get; set; }
    public decimal AmountPaid { get; set; }
    public decimal PenaltyApplied { get; set; }
    public FeeType FeeType { get; set; } = FeeType.Tuition;
    public decimal OutstandingBalance { get; set; }
    public PaymentStatus PaymentStatus { get; set; } = PaymentStatus.Unpaid;
    public PaymentType PaymentType { get; set; } = PaymentType.NotSet;
    public string? Notes { get; set; }

    public Candidate Candidate { get; set; } = default!;
    public Enrollment? Enrollment { get; set; }
    public ICollection<InstallmentPlan> InstallmentPlans { get; set; } = [];
    public ICollection<PaymentTransaction> PaymentTransactions { get; set; } = [];
    public Fine? Fine { get; set; }
}

File: frontend/zeal-edu/src/app/core/models/fee-structure.ts
import {
  FeeType,
  InstallmentStatus,
  PaymentMethod,
  PaymentStatus,
  PaymentType,
} from '@core/models/payment-enums';

export interface FeeStructureListItem {
  feeId: string;
  candidateId: string;
  candidateCode: string;
  candidateFullName: string;
  courseTitle: string | null;
  feeType: FeeType;
  totalFee: number;
  amountPaid: number;
  outstandingBalance: number;
  paymentStatus: PaymentStatus;
  paymentType: PaymentType;
  createdAt: string;
}

export interface InstallmentPlanItem {
  id: string;
  installmentNo: number;
  amountDue: number;
  dueDate: string;
  amountPaid: number;
  paidDate: string | null;
  status: InstallmentStatus;
  penaltyAmount: number;
}

export interface PaymentTransactionItem {
  id: string;
  installmentPlanId: string | null;
  installmentNo: number | null;
  receiptNumber: string;
  paymentDate: string;
  amount: number;
  paymentMethod: PaymentMethod;
  processedByStaffName: string | null;
  hasBankTransferProof: boolean;
}

export interface FeeStructureDetail {
  feeId: string;
  candidateId: string;
  candidateCode: string;
  candidateFullName: string;
  candidateEmail: string | null;
  candidatePhone: string | null;
  candidateIsActive: boolean;
  enrollmentId: string | null;
  enrollmentDate: string | null;
  courseId: string | null;
  courseName: string | null;
  durationWeeks: number | null;
  feeType: FeeType;
  totalFee: number;
  penaltyApplied: number;
  projectedPenalty: number;
  adjustedTotalFee: number;
  amountPaid: number;
  outstandingBalance: number;
  paymentStatus: PaymentStatus;
  paymentType: PaymentType;
  notes: string | null;
  installmentPlans: InstallmentPlanItem[];
  paymentTransactions: PaymentTransactionItem[];
}
